### PR TITLE
[Content] Unfettered Shipyard and Outfitter Updates

### PR DESCRIPTION
**Pull request by EjoThims**
**Content**

## Summary

Putting this up for @Hurleveur since he's not sure about the available time he'll have during his upcoming trip to make any suggested/needed changes.

This switches up Unfettered shipyards to sell slightly different variants on the ships they offer. The reason for this is twofold:

Firstly, they prefer not to sell the player builds that would work against Hai. They engage in their protest as they see fit, but when it comes to arming outsiders, they do so with the expectation it is primarily for fighting other outsiders. This means removing excessive generator output or battery space that tends to be favored in stock Hai builds as a buffer against ion damage.

Secondly, their resources are indeed limited, meaning they generally hold pieces they cannot manufacture themselves in reserve, while still selling what they can make to outsiders exactly to help alleviate that resource problem. This means generally not selling outfits on ships they don't sell outright.

These changes make it so that they do not sell Boulder Reactors to the player, and that bundled with the Shield Beetle is the only place they sell either ion cannons (down to two, though - while bumping it's weapon space usage to 332/333) or tracker pods. A smaller battery on the Shield Beetle as well in trade for more smaller reactors also plays up this angle, but the extra keystone is just because they need the extra luck.

This also makes the Unfettered fleets use their local variant of the Aphid (as their only dedicated cargo/transport ship), and adds the sand cell (which they sell on the Shield Beetle now and always have on the Aphid), asteroid scanner (which is needed for the mining jobs, since the stat is not otherwise found in Hai space), and outfit scanner (to complete the types, now that they offer the other 3) to the basic outfitter.

### CODE REVIEW

**Review by Hurleveur — APPROVED** [review]
Obviously since I did those changes I approve them ~~no conflict of interest there whatsoever~~

Just to detail my reasoning a bit here:
They do give you jammers on the ships, it only makes sense given you can buy them in the outfitter (here it's mostly a kind of "hey look we have better jammers than the other Hai")

Also they downgraded the reactors because they don't want to sell them to you except if they really have to. Even the Shield Beetle does not have a Boulder Reactor anymore, and the Lightning bug lost its Geode reactor.

The only moment when you get ion cannons is on the SB, and much less than before, also they don't sell you the biggest battery on it. This works into the lore I'm making for them: ion cannons, big batteries, big reactors... They realize those are limited, and they use them a lot, but they would not give them to the player just like that. They don't want to mass sell you tracker pods either, except on dedicated ships.
They have to use the ion cannons for prototypes, and currently whatever you do you are still +- a stranger. That will of course change with the campaign.

Overall this also goes into the philosophy of normal spaceport builds, that are not aimed at being specifically against ion by having extra energy, like the Hai ones are (and should remain, I think)
As for the Aphids if they're literally used to move cargo and ppl around and never get attacked by anyone, they may as well do it better with 30 more cargo and 2 crew in the build, at the (negligible) cost of that excess energy they used to have. Not only the player can use cargo expansions (idk about building it into the base hull, I think they just wouldn't waste resources on it)

File: data/hai/hai fleets.txt [hunk 1]
```diff
@@ -649,43 +649,43 @@ fleet "Unfettered Logistics"
...
-		"Aphid" 3
+		"Aphid (Unfettered)" 3
 		"Sea Scorpion (Pulse)"
 	variant 2
-		"Aphid" 4
+		"Aphid (Unfettered)" 4
 		"Sea Scorpion (Tracker)"
 	variant 2
-		"Aphid" 4
+		"Aphid""Aphid (Unfettered)" 4
```
Comment (Hurleveur) [inline data/hai/hai fleets.txt:676]:
```suggestion
		"Aphid (Unfettered)" 4
```

File: data/hai/hai ships.txt [hunk 2]
```diff
@@ -526,6 +539,22 @@ ship "Lightning Bug" "Lightning Bug (Surveillance)"
 		`"Benga" Atomic Steering`
 		"Hyperdrive"
 
+ship "Lightning Bug" "Lightning Bug (Unfettered Shipyards)"
+	outfits
+		"Pulse Cannon"
+		"Pulse Turret" 2
+		"Hai Corundum Regenerator"
+		"Hai Octopus Jammer"
```
Comment (Hurleveur) [inline data/hai/hai ships.txt:547]:
```suggestion
```

File: data/hai/hai ships.txt [hunk 3]
```diff
@@ -526,6 +539,22 @@ ship "Lightning Bug" "Lightning Bug (Surveillance)"
...
+		"Hai Williwaw Cooling"
+		"Quantum Keystone"
+		"Value Detector"
```
Comment (Hurleveur) [inline data/hai/hai ships.txt:550]:
```suggestion
		"Value Detector" 2
```

**Review by Hurleveur — COMMENTED** [review]
sorry looks like I left some small mistakes here
pretty easy to fix though

File: data/hai/hai fleets.txt (hunk 1 above)
Comment (EjoThims) [inline data/hai/hai fleets.txt:676]:
```suggestion
		"Aphid (Unfettered)" 4
```

File: data/hai/hai ships.txt [hunk 4]
```diff
@@ -64,6 +64,19 @@ ship "Aphid" "Aphid (Armed)"
 		`"Benga" Atomic Steering`
 		"Hyperdrive"
 
+ship "Aphid" "Aphid (Unfettered)"
+	outfits
```
Comment (EjoThims) [inline data/hai/hai ships.txt:68]:
```suggestion
	outfitss
```
test

**Comment by Hurleveur** [comment]
Oh and in case someone has a doubt: the prices are +- the same as before (maybe a few hundred thousands of difference at most, but really nothing that would matter)

**Review by Zitchas — APPROVED** [review]
Looks good to me. It makes complete sense that a faction wouldn't sell a stock ship that is built to fight against themselves or people that they care about. Nothing says that the shipyard sale version has to be the same as the version the faction typically uses.

In any case, the changes look good, and their lore reasoning is both appropriate and interesting.

**Comment by Luuiscool4567** [comment]
My only problem with this is that there are too many variants of the same ship that will  overfill the shipyard tab on the map.

**Comment by Hurleveur** [comment]
True, if we overdid it, but with just 3 additional ships it should be fine.

The Hai are pretty much the only ones that sell different versions of the same ship instead of selling only completely different ships like humans do.
Also if it truly becomes a problem I think a code change should be done instead of restricting this (like consider all ships like the default variant and dont show the other ones, or an option or smth else)

**Comment by Zitchas** [comment]
I don't see a problem with 3 variant ships in the shipyard. I wouldn't want adding lots of variants to become normalized, but there's a pretty clear reasoning behind the Unfettered's variants, and it makes sense.

That being said, there's a variety of options (particularly with code) that we could implement to improve things in the future. One notable option would be to limit the map shipyard list to only listing hulls.

**Review by MasterOfGrey — APPROVED** [review]

**Comment by MasterOfGrey** [comment]
> One notable option would be to limit the map shipyard list to only listing hulls.

That's something I'd like to see anyway tbh.

**Pull request merged**